CONTRIBUTING.md

```diff
@@ -78,6 +79,66 @@ use only the standard libraries, or the third-party packages we have decided to
 
 For rationale, check out the post [The Case Against Third Party Libraries](http://blog.gopheracademy.com/advent-2014/case-against-3pl/).
 
+### Writing unit tests
+Unit tests for Go code should be in the same package as the code itself.
+It should be in a file named `filename_test.go` with the package name `mypackage_test` as the name of the package.
+The tests should test functionality through public functions and methods.
+
+If access to internal members or functions is needed, then the test file should be named `filename_internal_test.go` and use the same package name as the package itself.
+The tests written in this file will have access to the internals of the package.
+This file can also be used to create testing-specific functions that can be exposed to the tests in `mypackage_test`.
+
+Pull requests require tests to be written for them and must pass our code coverage checks.
```

Comment:
This might be a good place to link to a longer article about writing end-to-end tests that includes a code sample, once we have that.